### PR TITLE
day_of_week

### LINKED ISSUE #9606: API: timeseries accessors naming convention
we have a slightly odd naming convention for several index (and .dt) accessors

`dayofweek`
`dayofyear`

(and adding `daysinmonth` in #9605)

so these should prob be `day_of_week` 

we should be able to maintain compat on these and just add the new ones.

### PR DESCRIPTION
**Pull request by huang-xiaowei32**
- [ ] closes #9606
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by mroeschke** [comment]
As mentioned https://github.com/pandas-dev/pandas/pull/25924#issuecomment-478132760, `dayofweek` needs to be deprecated first. This can't just be replaced.

File: asv_bench/benchmarks/timestamp.py [hunk 1]
```diff
@@ -44,7 +44,10 @@ def time_tz(self, tz, freq):
         self.ts.tz
 
     def time_dayofweek(self, tz, freq):
-        self.ts.dayofweek
+        self.ts.day_of_week
+
+    def time_day_of_week(self, tz, freq):
```
Comment (jreback) [inline asv_bench/benchmarks/timestamp.py]: indent error here

File: asv_bench/benchmarks/timestamp.py [hunk 2]
```diff
@@ -44,7 +44,10 @@ def time_tz(self, tz, freq):
         self.ts.tz
 
     def time_dayofweek(self, tz, freq):
```
Comment (jreback) [inline asv_bench/benchmarks/timestamp.py]: you can remove the original one

File: doc/redirects.csv [hunk 3]
```diff
@@ -527,7 +527,8 @@ generated/pandas.DatetimeIndex.ceil,../reference/api/pandas.DatetimeIndex.ceil
 generated/pandas.DatetimeIndex.date,../reference/api/pandas.DatetimeIndex.date
 generated/pandas.DatetimeIndex.day,../reference/api/pandas.DatetimeIndex.day
 generated/pandas.DatetimeIndex.day_name,../reference/api/pandas.DatetimeIndex.day_name
-generated/pandas.DatetimeIndex.dayofweek,../reference/api/pandas.DatetimeIndex.dayofweek
+generated/pandas.DatetimeIndex.dayofweek,../reference/api/pandas.DatetimeIndex.day_of_week
+generated/pandas.DatetimeIndex.day_of_week,../reference/api/pandas.DatetimeIndex.day_of_week
```
Comment (jreback) [inline doc/redirects.csv]: i don't think you need to add the new one, we don't have original pages for it

File: doc/source/reference/arrays.rst [hunk 4]
```diff
@@ -60,6 +60,7 @@ Properties
    Timestamp.asm8
    Timestamp.day
    Timestamp.dayofweek
```
Comment (jreback) [inline doc/source/reference/arrays.rst]: you can remove the original from the docs

File: doc/source/whatsnew/v0.16.2.rst [hunk 5]
```diff
@@ -138,7 +138,7 @@ Bug Fixes
 - Bug where ``Panel.from_dict`` does not set dtype when specified (:issue:`10058`)
 - Bug in ``Index.union`` raises ``AttributeError`` when passing array-likes. (:issue:`10149`)
 - Bug in ``Timestamp``'s' ``microsecond``, ``quarter``, ``dayofyear``, ``week`` and ``daysinmonth`` properties return ``np.int`` type, not built-in ``int``. (:issue:`10050`)
-- Bug in ``NaT`` raises ``AttributeError`` when accessing to ``daysinmonth``, ``dayofweek`` properties. (:issue:`10096`)
+- Bug in ``NaT`` raises ``AttributeError`` when accessing to ``daysinmonth``, ``dayofweek``, ``day_of_week`` properties. (:issue:`10096`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.2.rst]: don't change the original what's new notes

File: pandas/_libs/tslibs/period.pyx [hunk 6]
```diff
@@ -1957,6 +1957,57 @@ cdef class _Period(object):
...
+        >>> per.day_of_week
+        2
+        >>> per.end_time.day_of_week
+        2
+        """
+        base, mult = get_freq_code(self.freq)
+        return pweekday(self.ordinal, base)
+
     @property
     def dayofweek(self):
```
Comment (jreback) [inline pandas/_libs/tslibs/period.pyx]: just do

dayofweek = day_of_week

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you merge master and update per comments

**Comment by jreback** [comment]
closing as stale. this is pretty close, so ping if you'd like to continue.

**Pull request closed without merging**